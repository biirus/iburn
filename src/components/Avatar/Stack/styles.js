export default ({ spacing, colors }) => {
  return {
    /* Styles applied to root element */
    root: {
      display: "flex",
      lineHeight: 1,
    },

    /* Styles applied to the wrapper of `Avatar` element */
    avatarWrapper: {
      position: "relative",
    },

    /* Styles applied to each `Avatar` element */
    avatar: {
      border: `2px solid ${colors.n0}`,
    },
  };
};
