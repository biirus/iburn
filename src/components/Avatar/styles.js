export default ({ colors, typography, spacing, }) => {
  const size = spacing.unit * 6;

  return {
    root: {
      ...typography.h600,
      color: colors.w700a,

      display: "inline-flex",
      alignItems: "center",
      justifyContent: "center",
      alignSelf: "baseline",
      overflow: "hidden",
      width: size,
      height: size,

      backgroundColor: colors.n100,
      borderRadius: "50%",
    },

    img: {
      width: "100%",
      height: "100%",
      objectFit: "cover",
    },
  };
};
