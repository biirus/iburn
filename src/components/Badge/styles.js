export default ({ spacing, palette, colors, borderRadius }) => {
  return {
    /* Styles applied to the root element */
    root: {
      display: "inline-flex",
      padding: [[spacing.small / 2, spacing.small]],
      lineHeight: 1,
      borderRadius,
    },

    /* Styles applied to the root element if `color="default"` */
    default: {
      backgroundColor: colors.n20a,
      color: colors.n400,
    },

    /* Styles applied to the root element if `color="primary"` */
    primary: {
      backgroundColor: palette.primary,
      color: palette.text.invert,
    },

    /* Styles applied to the root element if `color="error"` */
    error: {
      backgroundColor: palette.error,
      color: palette.text.invert,
    },

    /* Styles applied to the root element if `color="success"` */
    success: {
      backgroundColor: palette.success,
      color: palette.text.invert,
    },

    /* Styles applied to the root element if `color="warning"` */
    warning: {
      backgroundColor: palette.warning,
      color: palette.text.invert,
    },

    /* Styles applied to the root element if `color="info"` */
    info: {
      backgroundColor: palette.info,
      color: palette.text.invert,
    },
  };
};
