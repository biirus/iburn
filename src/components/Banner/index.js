import React from "react";
import cn from "classnames";
import PropTypes from "prop-types";

// assets
import Icon from "components/Icon";

// enhancers
import { enhance } from "./enhance";

/**
 * Component displays a prominent banner message and icon.
 * It is designed to display a message at the top of the page.
 * Banner animates its opening and closing.
 */
class Banner extends React.Component {
  static propTypes = {
    /**
     * Content to be shown next to the icon
     */
    children: PropTypes.node,
    /**
     * Class name string to be merged to the root node
     */
    className: PropTypes.string,
    /**
     * [JSS](http://cssinjs.org/react-jss/) classes object notation
     */
    classes: PropTypes.object,
    /**
     * A custom icon element to be shown left of the main content
     */
    icon: PropTypes.node,
    /**
     * The Banner background color with default icon
     */
    color: PropTypes.oneOf(["warning", "success", "error", "info", "default"]),
    /**
     * Delay before calling `onClose` prop
     */
    autoCloseTimeout: PropTypes.number,
    /**
     * Callback fired when the component requests to be closed. Typically onClose is used to set state in the parent component, which is used to control the Banner's `isOpen` prop.
     */
    onClose: PropTypes.func,
    /**
     * Callback fired when `Banner` is opening
     */
    onEnter: PropTypes.func,
    /**
     * Callback fired when `Banner` is closing
     */
    onExit: PropTypes.func,
  };

  componentDidUpdate (oldProps) {
    const { isOpen, onEnter, onExit } = this.props;

    const isOpening = !oldProps.isOpen && isOpen;
    const isClosing = oldProps.isOpen && !isOpen;

    if (isOpening) {
      onEnter && onEnter();
      this.startTimeout();
    }

    if (isClosing) {
      this.endTimeout();
      onExit && onExit();
    }
  }

  componentWillUnmount () {
    this.endTimeout();
  }

  startTimeout () {
    const { autoCloseTimeout, onClose } = this.props;

    if (autoCloseTimeout) {
      if (this.timeout) {
        clearTimeout(this.timeout);
      }

      this.timeout = setTimeout(() => {
        onClose && onClose("timeout");
      }, autoCloseTimeout);
    }
  }

  endTimeout () {
    if (this.timeout) {
      clearTimeout(this.timeout);
    }
  }

  render () {
    const {
      icon,
      color,
      isOpen,
      classes,
      children,
      className,
      autoCloseTimeout,
      onClose,
      onEnter,
      onExit,
      theme,
      ...rest
    } = this.props;

    const classNames = cn(classes.root, className, classes[color], {
      [classes.open]: isOpen,
    });

    return (
      <div {...rest} className={classNames}>
        {(icon || (color && color !== "default")) && (
          <span className={classes.icon}>
            {icon || <Icon label={`editor/${color}`} />}
          </span>
        )}
        <span className={classes.label}>{children}</span>
      </div>
    );
  }
}

export default enhance(Banner);
