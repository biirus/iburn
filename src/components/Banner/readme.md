```js
class Example extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      default: false,
      warning: false,
      error: false,
      success: false,
      info: false
    };

    this.toggle = this.toggle.bind(this);
  }

  toggle(e) {
    const { name } = e.currentTarget;
    this.setState(state => ({ [name]: !state[name] }));
  }

  render() {
    const banners = ["default", "warning", "error", "success", "info"];

    return (
      <div>
        {banners.map(banner => (
          <Button
            key={banner}
            name={banner}
            color={banner}
            onClick={this.toggle}
            style={{ margin: 8 }}
          >
            Toggle {banner}
          </Button>
        ))}

        {banners.map(banner => (
          <Banner
            key={banner}
            isOpen={this.state[banner]}
            color={banner}
            icon={banner === "default" && <Icon label="lock" />}
          >
            {banner} banner content
          </Banner>
        ))}
      </div>
    );
  }
}
<Example />;
```
