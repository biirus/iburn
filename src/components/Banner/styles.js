export default ({
  spacing,
  palette,
  shadows,
  typography,
  zIndex,
  transitions,
}) => {
  const iconSize = spacing.unit * 3;

  return {
    root: {
      ...typography.h100,

      position: "fixed",
      top: 0,
      left: 0,
      width: "100%",
      zIndex: zIndex.flag,

      display: "flex",
      alignItems: "center",
      justifyContent: "center",
      padding: spacing.medium,

      boxShadow: shadows.small,
      backgroundColor: palette.backgrounds.default,
      color: palette.text.primary,

      opacity: 0,
      transform: `translateY(-100%)`,

      transition: transitions.create(["opacity", "transform",], "leavingScreen"),
    },

    open: {
      opacity: 1,
      transform: `translateY(0)`,

      transition: transitions.create(
        ["opacity", "transform",],
        "enteringScreen"
      ),
    },

    warning: {
      backgroundColor: palette.messages.warning,
      color: palette.text.invert,

      "& $icon": {
        fill: palette.messages.warning,
      },
    },

    error: {
      backgroundColor: palette.messages.error,
      color: palette.text.invert,

      "& $icon": {
        fill: palette.messages.error,
      },
    },

    info: {
      backgroundColor: palette.messages.info,
      color: palette.text.invert,

      "& $icon": {
        fill: palette.messages.info,
      },
    },

    success: {
      backgroundColor: palette.messages.success,
      color: palette.text.invert,

      "& $icon": {
        fill: palette.messages.success,
      },
    },

    icon: {
      flex: `0 0 ${iconSize}px`,
      lineHeight: 0,
    },

    label: {
      flex: `0 1 auto`,
      marginLeft: spacing.small / 2,
      overflow: "hidden",
      textOverflow: "ellipsis",
      whiteSpace: "nowrap",
      lineHeight: `${iconSize}px`,
    },
  };
};
