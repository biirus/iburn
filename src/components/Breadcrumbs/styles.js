export default theme => {
  return {
    /* Styles applied to the root element. */
    root: {
      overflow: "hidden",
      whiteSpace: "nowrap",
      textOverflow: "ellipsis",
      color: theme.palette.text.secondary,
    },

    /* Styles applied to the item's `span` element */
    item: {
      "&+$item::before": {
        content: `"/"`,
        display: "inline-block",
        width: theme.spacing.medium,
        textAlign: "center",
      },
    },

    /* Styles applied to the item's `a` element */
    link: {
      color: theme.palette.text.secondary,
      textDecoration: "none",

      "&:hover": {
        textDecoration: "underline",
      },
    },
  };
};
