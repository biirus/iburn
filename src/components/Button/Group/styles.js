export default ({ borderRadius, }) => ({
  root: {
    display: "inline-flex",
  },

  justified: {
    width: "100%",

    "& $item": {
      flex: "1 0 auto",
      width: 0,

      "& $inner": {
        width: "100%",
        textAlign: "center",
      },
    },
  },

  item: {
    flex: "1 0 auto",

    "& + $item": {
      marginLeft: 1,
    },

    "&:not(:first-child):not(:last-child) $inner": {
      borderRadius: 0,
    },

    "&:first-child $inner": {
      borderRadius: [borderRadius, 0, 0, borderRadius,],
    },

    "&:last-child $inner": {
      borderRadius: [0, borderRadius, borderRadius, 0,],
    },
  },

  inner: {},
});
