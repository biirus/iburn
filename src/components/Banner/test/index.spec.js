import React from "react";

import Icon from "../../Icon";
import Banner from "../index";
import styles from "../styles";

const classes = getClasses(styles);
const Component = Banner.InnerComponent;

describe("<Banner />", () => {
  it("should merge user classes", () => {
    const wrapper = shallow(
      <Component className="customClass" classes={classes}>
        Banner content
      </Component>
    );

    expect(wrapper.prop("className")).toEqual("root customClass");
  });

  it("should render a div with only `label` child", () => {
    const wrapper = shallow(
      <Component classes={classes}>Banner content</Component>
    );

    expect(wrapper.name()).toEqual("div");
    expect(wrapper.children().length).toEqual(1);
    expect(wrapper.childAt(0).prop("className")).toEqual("label");
  });

  it("should render a div with `warning` color <Icon /> and `label` children", () => {
    const wrapper = shallow(
      <Component color="warning" classes={classes}>
        Banner content
      </Component>
    );

    expect(wrapper.name()).toEqual("div");
    expect(wrapper.children().length).toEqual(2);
    expect(wrapper.find("Icon").exists()).toBeTruthy();
    expect(wrapper.prop("className")).toEqual("root warning");
    expect(wrapper.childAt(1).text()).toEqual("Banner content");
  });

  it("should render a div with custom icon and `label` children", () => {
    const wrapper = shallow(
      <Component icon={<Icon label="lock" />} classes={classes}>
        Banner content
      </Component>
    );

    expect(wrapper.name()).toEqual("div");
    expect(wrapper.children().length).toEqual(2);
    expect(wrapper.childAt(0).contains(<Icon label="lock" />)).toBeTruthy();
    expect(wrapper.childAt(1).text()).toEqual("Banner content");
  });

  it("should render a Banner with `color` prop correctly", () => {
    const wrapper = shallow(
      <Component color="warning" classes={classes}>
        Banner content
      </Component>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("should render a Banner with custom icon correctly", () => {
    const wrapper = shallow(
      <Component icon={<Icon label="lock" />} classes={classes}>
        Banner content
      </Component>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("should render a Banner with only label correctly", () => {
    const wrapper = shallow(
      <Component classes={classes}>Banner content</Component>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
