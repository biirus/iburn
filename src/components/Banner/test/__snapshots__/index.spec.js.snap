// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner /> should render a Banner with \`color\` prop correctly 1`] = `
<div
  className="root warning"
>
  <span
    className="icon"
  >
    <Icon
      label="editor/warning"
    />
  </span>
  <span
    className="label"
  >
    Banner content
  </span>
</div>
`;

exports[`<Banner /> should render a Banner with custom icon correctly 1`] = `
<div
  className="root"
>
  <span
    className="icon"
  >
    <Icon
      label="lock"
    />
  </span>
  <span
    className="label"
  >
    Banner content
  </span>
</div>
`;

exports[`<Banner /> should render a Banner with only label correctly 1`] = `
<div
  className="root"
>
  <span
    className="label"
  >
    Banner content
  </span>
</div>
`;
